base/version_comparable.go

```diff
@@ -129,10 +137,25 @@ func (a *ComparableVersion) Less(b *ComparableVersion) bool {
 	return false
 }
 
-func (pv *ComparableVersion) String() string {
-	pv.strOnce.Do(func() {
-		pv.str = pv.formatComparableVersion()
-	})
+// AtLeastMinorDowngrade returns true there is a major or minor downgrade from a to b.
+func (a *ComparableVersion) AtLeastMinorDowngrade(b *ComparableVersion) bool {
```

Comment:
Possibly a nitpick, but I thought this would be easier to follow in the form:
```
if a.epoch != b.epoch {
    return a.epoch > b.epoch
}
etc
```
It feels a bit more transparent that if the values are equal, we fall through to the next check.  If you prefer the way it is I can live with that.